[Guided onboarding] Add API integration tests

mibragimov opened the pull request:
Fixes https://github.com/elastic/kibana/issues/142261

### Summary

This PR adds the following API integration tests for the guided onboarding plugin:

 #### GET /state
- [x]  Returns the state for all guides (?active=false)
- [x]  Returns state for the active guide (?active=true)
- [x]  Returns an empty array if saved object doesn't exist

#### PUT /state

- [x] Updates a guide that has an existing saved object
- [x] Updates a guide that does not have a saved object
- [x] Updates any existing active guides to inactive

alisonelizabeth commented on x-pack/test/api_integration/apis/guided_onboarding/get_state.ts:
```diff
@@ -0,0 +1,81 @@
+      expect(response.body).to.eql({ state: [testGuideA] });
+    });
+
+    it('should return state from saved object if it exists', async () => {
```
I think this test case is covered by `should return the state for all guides` but let me know if I missed something 😅

alisonelizabeth commented on x-pack/test/api_integration/apis/guided_onboarding/put_state.ts:
```diff
@@ -0,0 +1,126 @@
+      expect(updatedGuide.status).to.eql('ready_to_complete');
+    });
+
+    it('should update any existing active guides to inactive', async () => {
```
Re: my comment above. I see this test case would not be possible without using a second/third guide ID.

alisonelizabeth commented on x-pack/test/api_integration/apis/guided_onboarding/get_state.ts:
```diff
@@ -0,0 +1,81 @@
+import type { FtrProviderContext } from '../../ftr_provider_context';
+
+const testGuideA = testGuideStep1ActiveState;
+const testGuideB = { ...testGuideNotActiveState, guideId: 'testGuideB' as GuideId }; // different guideId
```
We could also use one of search/observability/security guideIds. For API integration tests, I think it's OK to use these. @yuliacech WDYT?

alisonelizabeth reviewed (commented):
Nice work @muhammad-elastic! These tests are looking great. 

I have some concerns with using an unsupported `guideId` in the tests. @yuliacech curious on your thoughts about this too. Also, were you able to run these through the flaky test runner? Thanks!

mibragimov commented:
> Nice work @muhammad-elastic! These tests are looking great.
> 
> I have some concerns with using an unsupported `guideId` in the tests. @yuliacech curious on your thoughts about this too. Also, were you able to run these through the flaky test runner? Thanks!

Thanks for the review @alisonelizabeth! Addressed the comments with https://github.com/elastic/kibana/pull/144104/commits/91b56aaa7ba6b2f7f000bbc4874f4ac1b8090792 and ran the tests through the flaky test runner ~200 times https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1530#_

yuliacech reviewed (approved):
Thanks a lot for adding the tests, @mibragimov! 
I agree with @alisonelizabeth to use only supported `guideId` values in the tests (testGuide, observability, search, security). It reflects the code better but we also might want to add a check for the supported IDs in the API later.

alisonelizabeth reviewed (approved):
Thanks for addressing my feedback! Latest LGTM.

elasticmachine commented:
Pinging @elastic/platform-onboarding (Team:Journey/Onboarding)
